diff -urN a/arch/arm/dts/px30-evb.dts b/arch/arm/dts/px30-evb.dts
--- a/arch/arm/dts/px30-evb.dts	2023-09-20 19:58:32.504450853 +0900
+++ b/arch/arm/dts/px30-evb.dts	2023-09-20 19:58:55.416193991 +0900
@@ -406,7 +406,7 @@
 &emmc {
 	bus-width = <8>;
 	cap-mmc-highspeed;
-	mmc-hs200-1_8v;
+	mmc-ddr-1_8v;
 	supports-emmc;
 	disable-wp;
 	non-removable;
diff -urN a/configs/px30_defconfig b/configs/px30_defconfig
--- a/configs/px30_defconfig	2023-09-20 19:58:32.568450136 +0900
+++ b/configs/px30_defconfig	2023-09-20 19:58:55.416193991 +0900
@@ -41,6 +41,7 @@
 # CONFIG_CMD_FLASH is not set
 # CONFIG_CMD_FPGA is not set
 CONFIG_CMD_GPT=y
+CONFIG_CMD_GPIO=y
 # CONFIG_CMD_LOADB is not set
 # CONFIG_CMD_LOADS is not set
 CONFIG_CMD_BOOT_ANDROID=y
diff -urN a/drivers/usb/host/ehci-hcd.c b/drivers/usb/host/ehci-hcd.c
--- a/drivers/usb/host/ehci-hcd.c	2023-09-20 19:58:32.600449778 +0900
+++ b/drivers/usb/host/ehci-hcd.c	2023-09-20 19:59:42.755664000 +0900
@@ -1683,7 +1683,7 @@
 
 	ret = generic_phy_get_by_index(dev, index, phy);
 	if (ret) {
-		if (ret != -ENOENT) {
+		if (ret != -ENOENT && ret != -ENODEV) {
 			dev_err(dev, "failed to get usb phy\n");
 			return ret;
 		}
diff -urN a/drivers/usb/host/ohci-generic.c b/drivers/usb/host/ohci-generic.c
--- a/drivers/usb/host/ohci-generic.c	2023-09-20 19:58:32.600449778 +0900
+++ b/drivers/usb/host/ohci-generic.c	2023-09-20 20:00:24.355199015 +0900
@@ -33,7 +33,7 @@
 
 	ret = generic_phy_get_by_index(dev, index, &priv->phy);
 	if (ret) {
-		if (ret != -ENOENT) {
+		if (ret != -ENOENT && ret != -ENODEV) {
 			dev_err(dev, "failed to get usb phy\n");
 			return ret;
 		}
diff -urN a/make.sh b/make.sh
--- a/make.sh	2023-09-20 19:58:32.632449418 +0900
+++ b/make.sh	2023-09-20 19:58:55.416193991 +0900
@@ -11,7 +11,7 @@
 CMD_ARGS=$1
 
 ########################################### User can modify #############################################
-RKBIN_TOOLS=../rkbin/tools
+RKBIN_TOOLS=$ETC_DIR/rockchip/rkbin/tools
 CROSS_COMPILE_ARM32=../prebuilts/gcc/linux-x86/arm/gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf/bin/arm-linux-gnueabihf-
 CROSS_COMPILE_ARM64=../prebuilts/gcc/linux-x86/aarch64/gcc-linaro-6.3.1-2017.05-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
 ########################################### User not touch #############################################
diff -urN a/scripts/uboot.sh b/scripts/uboot.sh
--- a/scripts/uboot.sh	2023-09-20 19:58:32.632449418 +0900
+++ b/scripts/uboot.sh	2023-09-20 19:58:55.416193991 +0900
@@ -50,6 +50,6 @@
 	exit 1
 fi
 
-../rkbin/tools/loaderimage --pack --uboot u-boot.bin uboot.img ${LOAD_ADDR} ${SIZE}
+$ETC_DIR/rockchip/rkbin/tools/loaderimage --pack --uboot u-boot.bin uboot.img ${LOAD_ADDR} ${SIZE}
 echo "pack uboot okay! Input: u-boot.bin"
 echo
