diff -urN a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dtsi b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dtsi
--- a/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dtsi	2023-09-25 11:08:17.576862016 +0900
+++ b/arch/arm64/boot/dts/rockchip/px30-evb-ddr3-v10.dtsi	2023-09-25 11:08:35.816683595 +0900
@@ -93,11 +93,11 @@
 		rockchip,android-charge-on = <1>;
 		rockchip,uboot-low-power-voltage = <3500>;
 		rockchip,screen-on-voltage = <3600>;
-		status = "okay";
+		status = "disabled";
 	};
 
 	rk809_sound: rk809-sound {
-		status = "okay";
+		status = "disabled";
 		compatible = "rockchip,multicodecs-card";
 		rockchip,card-name = "rockchip-rk809";
 		hp-det-gpio = <&gpio2 RK_PB0 GPIO_ACTIVE_LOW>;
@@ -135,7 +135,7 @@
 	};
 
 	test-power {
-		status = "okay";
+		status = "disabled";
 	};
 
 	vcc_phy: vcc-phy-regulator {
@@ -159,7 +159,7 @@
 		wifi_chip_type = "AP6210";
 		WIFI,host_wake_irq = <&gpio0 RK_PB2 GPIO_ACTIVE_HIGH>;
 		WIFI,poweren_gpio = <&gpio0 RK_PA2 GPIO_ACTIVE_HIGH>;
-		status = "okay";
+		status = "disabled";
 	};
 
 	wireless_bluetooth: wireless-bluetooth {
@@ -173,16 +173,16 @@
 		BT,reset_gpio = <&gpio0 RK_PC1 GPIO_ACTIVE_HIGH>;
 		BT,wake_gpio = <&gpio0 RK_PA1 GPIO_ACTIVE_HIGH>;
 		BT,wake_host_irq = <&gpio0 RK_PB3 GPIO_ACTIVE_HIGH>;
-		status = "okay";
+		status = "disabled";
 	};
 };
 
 &display_subsystem {
-	status = "okay";
+	status = "disabled";
 };
 
 &dsi_in_vopb {
-	status = "okay";
+	status = "disabled";
 };
 
 &dsi_in_vopl {
@@ -191,7 +191,7 @@
 
 &route_dsi {
 	connect = <&vopb_out_dsi>;
-	status = "okay";
+	status = "disabled";
 };
 
 &bus_apll {
@@ -226,20 +226,16 @@
 
 &gmac {
 	phy-supply = <&vcc_phy>;
-	clock_in_out = "input";
-	assigned-clocks = <&cru SCLK_GMAC>;
-	assigned-clock-parents = <&gmac_clkin>;
-	pinctrl-names = "default";
-	pinctrl-0 = <&rmii_pins &mac_refclk>;
+	clock_in_out = "output";
 	snps,reset-gpio = <&gpio2 13 GPIO_ACTIVE_LOW>;
 	snps,reset-active-low;
 	snps,reset-delays-us = <0 50000 50000>;
-	status = "okay";
+	status = "okay";	
 };
 
 &gpu {
 	mali-supply = <&vdd_logic>;
-	status = "okay";
+	status = "disabled";
 };
 
 &i2c0 {
@@ -275,7 +271,7 @@
 		vcc9-supply = <&vcc5v0_sys>;
 
 		pwrkey {
-			status = "okay";
+			status = "disabled";
 		};
 
 		pinctrl_rk8xx: pinctrl_rk8xx {
@@ -504,7 +500,7 @@
 			pinctrl-0 = <&i2s1_2ch_mclk>;
 			hp-volume = <20>;
 			spk-volume = <3>;
-			status = "okay";
+			status = "disabled";
 		};
 	};
 };
@@ -545,7 +541,7 @@
 };
 
 &i2c2 {
-	status = "okay";
+	status = "disabled";
 
 	clock-frequency = <100000>;
 
@@ -576,7 +572,7 @@
 };
 
 &i2s1_2ch {
-	status = "okay";
+	status = "disabled";
 	#sound-dai-cells = <0>;
 };
 
@@ -591,11 +587,11 @@
 };
 
 &isp_mmu {
-	status = "okay";
+	status = "disabled";
 };
 
 &mipi_dphy_rx0 {
-	status = "okay";
+	status = "disabled";
 
 	ports {
 		#address-cells = <1>;
@@ -627,7 +623,7 @@
 };
 
 &nandc0 {
-	status = "okay";
+	status = "disabled";
 };
 
 &pinctrl {
@@ -640,7 +636,30 @@
 	pmic {
 		pmic_int: pmic_int {
 			rockchip,pins =
-				<0 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up>;
+				<0 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up>,
+				// DOUT
+				<1 RK_PB5 RK_FUNC_GPIO &pcfg_pull_none>,
+				<1 RK_PB6 RK_FUNC_GPIO &pcfg_pull_none>,
+				<1 RK_PB7 RK_FUNC_GPIO &pcfg_pull_none>,
+				// SW1
+				<3 RK_PC0 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PC1 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PC2 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PC3 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PC4 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PC5 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PC6 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PC7 RK_FUNC_GPIO &pcfg_pull_up>,
+				// SW2
+				<3 RK_PD0 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PD1 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PD2 RK_FUNC_GPIO &pcfg_pull_up>,
+				<3 RK_PD3 RK_FUNC_GPIO &pcfg_pull_up>,
+				// SW3
+				<1 RK_PD6 RK_FUNC_GPIO &pcfg_pull_up>,
+				<1 RK_PD7 RK_FUNC_GPIO &pcfg_pull_up>,
+				// LED
+				<1 RK_PB4 RK_FUNC_GPIO  &pcfg_pull_up>;
 		};
 
 		soc_slppin_gpio: soc_slppin_gpio {
@@ -674,15 +693,15 @@
 };
 
 &pwm1 {
-	status = "okay";
+	status = "disabled";
 };
 
 &rk_rga {
-	status = "okay";
+	status = "disabled";
 };
 
 &rkisp1 {
-	status = "okay";
+	status = "disabled";
 
 	port {
 		#address-cells = <1>;
@@ -696,12 +715,12 @@
 };
 
 &rockchip_suspend {
-	status = "okay";
+	status = "disabled";
 	rockchip,sleep-debug-en = <1>;
 };
 
 &saradc {
-	status = "okay";
+	status = "disabled";
 	vref-supply = <&vcc1v8_soc>;
 };
 
@@ -720,7 +739,7 @@
 	sd-uhs-sdr104;
 	vqmmc-supply = <&vccio_sd>;
 	vmmc-supply = <&vcc_sd>;
-	status = "okay";
+	status = "disabled";
 };
 
 &sdio {
@@ -740,13 +759,45 @@
 	pinctrl-names = "gpio", "otpout";
 	pinctrl-0 = <&tsadc_otp_pin>;
 	pinctrl-1 = <&tsadc_otp_out>;
-	status = "okay";
+	status = "disabled";
+};
+
+&uart0 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&uart0_xfer>;
+    status = "okay";
 };
 
 &uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart1_xfer &uart1_cts>;
-	status = "okay";
+    pinctrl-names = "default";
+    pinctrl-0 = <&uart1_xfer>;
+    status = "okay";
+};
+
+&uart3 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&uart3m0_xfer>;
+    status = "okay";
+};
+
+&uart4 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&uart4_xfer>;
+    status = "okay";
+};
+
+&uart5 {
+    pinctrl-names = "default";
+	pinctrl-0 = <&uart5_xfer>;
+    status = "okay";
+};
+
+&i2c3 {
+	status = "disabled";
+};
+
+&spi0 {
+	status = "disabled";
 };
 
 &u2phy {
diff -urN a/arch/arm64/boot/dts/rockchip/px30.dtsi b/arch/arm64/boot/dts/rockchip/px30.dtsi
--- a/arch/arm64/boot/dts/rockchip/px30.dtsi	2023-09-25 11:08:17.580861976 +0900
+++ b/arch/arm64/boot/dts/rockchip/px30.dtsi	2023-09-25 11:12:25.002249082 +0900
@@ -675,8 +675,7 @@
 		clocks = <&pmucru SCLK_UART0_PMU>, <&pmucru PCLK_UART0_PMU>;
 		clock-names = "baudclk", "apb_pclk";
 		dmas = <&dmac 0>, <&dmac 1>;
-		/*You can add it to enable dma*/
-		/*dma-names = "tx", "rx";*/
+		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
 		pinctrl-names = "default";
@@ -879,8 +878,7 @@
 		clocks = <&cru SCLK_UART1>, <&cru PCLK_UART1>;
 		clock-names = "baudclk", "apb_pclk";
 		dmas = <&dmac 2>, <&dmac 3>;
-		/*You can add it to enable dma*/
-		/*dma-names = "tx", "rx";*/
+		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
 		pinctrl-names = "default";
@@ -895,8 +893,7 @@
 		clocks = <&cru SCLK_UART2>, <&cru PCLK_UART2>;
 		clock-names = "baudclk", "apb_pclk";
 		dmas = <&dmac 4>, <&dmac 5>;
-		/*You can add it to enable dma*/
-		/*dma-names = "tx", "rx";*/
+		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
 		pinctrl-names = "default";
@@ -911,8 +908,7 @@
 		clocks = <&cru SCLK_UART3>, <&cru PCLK_UART3>;
 		clock-names = "baudclk", "apb_pclk";
 		dmas = <&dmac 6>, <&dmac 7>;
-		/*You can add it to enable dma*/
-		/*dma-names = "tx", "rx";*/
+		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
 		pinctrl-names = "default";
@@ -927,8 +923,7 @@
 		clocks = <&cru SCLK_UART4>, <&cru PCLK_UART4>;
 		clock-names = "baudclk", "apb_pclk";
 		dmas = <&dmac 8>, <&dmac 9>;
-		/*You can add it to enable dma*/
-		/*dma-names = "tx", "rx";*/
+		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
 		pinctrl-names = "default";
@@ -943,8 +938,7 @@
 		clocks = <&cru SCLK_UART5>, <&cru PCLK_UART5>;
 		clock-names = "baudclk", "apb_pclk";
 		dmas = <&dmac 10>, <&dmac 11>;
-		/*You can add it to enable dma*/
-		/*dma-names = "tx", "rx";*/
+		dma-names = "tx", "rx";
 		reg-shift = <2>;
 		reg-io-width = <4>;
 		pinctrl-names = "default";
